### DISCUSSION TITLE
Is there a way to exclude dependencies from the bundle?

### DISCUSSION THREAD

**Discussion by Blatty** [vercel/vercel#5108] [Help] [answered]
Hello,

Thanks for this amazing project & platform!

Is there a way to exclude specific Node dependencies that are only used at build time?

e.g:

```bash
node_modules/
  nexus-plugin-prisma
    @prisma/cli
    @prisma/client
    @prisma/sdk
```

I'd like to exclude `@prisma/cli` & `@prisma/sdk` from the bundle as it's not being used during runtime.

**Comment by styfle** [comment, marked as answer]
Hi @Blatty 

> I'd like to exclude @prisma/cli & @prisma/sdk from the bundle as it's not being used during runtime.

This is already the default behavior. The deployment process for Serverless Functions is something like this:

- Download source code
- Run `yarn install` or `npm install`
- Statically analyze each function in the `/api` directory to map out import/require
- Use import maps to bundle relevant files into each function

If you're seeing `@prisma/sdk` being bundled, its because there is something calling `require('@prisma/sdk')` or `import foo from '@prisma/sdk'`

**Reply by Blatty** [reply]
Thanks @styfle for your quick answer! So even if `@prisma/sdk` is not used  in my code but is a dependency of nexus-plugin-prisma, it will not be bundled?

Is there any ways to debug the modules tree & simulate a vercel build?

**Reply by styfle** [reply]
It depends on your source file. If your source file does `require('nexus-plugin-prisma')` and nexus-plugin-prisma does `require('@prisma/sdk')`, then @prisma/sdk will be included.

You can see most of the bundle behavior with nft:

```sh
mkdir example
cd example
npm init -y
npm add nexus-plugin-prisma
echo "require('nexus-plugin-prisma')" > index.js
npx @vercel/nft print index.js
```
